plugins/modules/remote_management/redfish/ilo_redfish_config.py

```diff
@@ -0,0 +1,178 @@
+    type: str
+    version_added: '0.2.0'
+requirements:
+    - "python >= 3.8"
+    - "ansible >= 3.2"
+author:
+    - "Bhavya B (@bhavya06)"
+'''
+
+EXAMPLES = '''
+  - name: Disable WINS Registration
+    community.general.ilo_redfish_config:
+      category: Manager
+      command: SetWINSReg
+      baseuri: "{{ baseuri }}"
+      username: "{{ username }}"
+      password: "{{ password }}"
+      attribute_name: "{{ attribute_name }}"
+
+  - name: Set Time Zone
+    community.general.ilo_redfish_config:
+      category: Manager
+      command: SetTimeZone
+      baseuri: "{{ baseuri }}"
+      username: "{{ username }}"
+      password: "{{ password }}"
+      attribute_name: TimeZone
+      attribute_value: "{{ attribute_value }}"
+'''
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'metadata_version': '1.1'}
+
+
+CATEGORY_COMMANDS_ALL = {
+    "Manager": ["SetTimeZone", "SetDNSserver", "SetDomainName", "SetNTPServers", "SetWINSReg"]
+}
+
+from ansible.module_utils._text import to_native
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.community.general.plugins.module_utils.ilo_redfish_utils import iLORedfishUtils
+
+
+def main():
+    result = {}
+    module = AnsibleModule(
+        argument_spec=dict(
+            category=dict(required=True),
+            command=dict(required=True, type='list', elements='str'),
+            baseuri=dict(required=True),
+            username=dict(),
+            password=dict(no_log=True),
+            attribute_name=dict(default=None),
+            attribute_value=dict(default=None),
+            auth_token=dict(no_log=True),
+            timeout=dict(type='int', default=10)
+        ),
+        supports_check_mode=False
+    )
+
+    category = module.params['category']
+    command_list = module.params['command']
+
+    # creds = None
+    session_id = None
+    creds = {"user": module.params['username'],
+             "pswd": module.params['password'],
+             "token": module.params['auth_token']}
+    # creds["Hpe"] = True
+
+    timeout = module.params['timeout']
+
+    if module.params['baseuri'] is None:
+        module.fail_json(msg=to_native("Baseuri is empty."))
+
+    root_uri = "https://" + module.params['baseuri']
+    # Auto_login = False
+    rf_utils = iLORedfishUtils(creds, root_uri, timeout, module)
+    mgr_attributes = {'mgr_attr_name': module.params['attribute_name'],
+                      'mgr_attr_value': module.params['attribute_value']}
+
+    if category not in CATEGORY_COMMANDS_ALL:
+        module.fail_json(msg=to_native("Invalid Category '%s'. Valid Categories = %s" % (
+            category, CATEGORY_COMMANDS_ALL.keys())))
+
+    for cmd in command_list:
+        if cmd not in CATEGORY_COMMANDS_ALL[category]:
+            module.fail_json(msg=to_native("Invalid Command '%s'. Valid Commands = %s" % (
+                cmd, CATEGORY_COMMANDS_ALL[category])))
```

Comment:
As an alternative, you could:
```python
    offending = [cmd for cmd in command_list if cmd not in CATEGORY_COMMANDS_ALL[category]]
    if offending:
        module.fail_json(msg=to_native("Invalid Command(s): '%s'. Allowed Commands = %s" % (offending, CATEGORY_COMMANDS_ALL[category])))
```
Slightly leaner and with the added advantage of returning all the offending commands, not just the first one.